scrypto/src/crypto_utils/crypto_utils.rs

```diff
@@ -87,11 +88,60 @@ impl CryptoUtils {
     }
 
     /// Calculates Keccak-256 digest over given vector of bytes
-    pub fn keccak256_hash(data: Vec<u8>) -> Hash {
+    pub fn keccak256_hash(data: impl AsRef<[u8]>) -> Hash {
         let hash = copy_buffer(unsafe {
-            crypto_utils::crypto_utils_keccak256_hash(data.as_ptr(), data.len())
+            crypto_utils::crypto_utils_keccak256_hash(data.as_ref().as_ptr(), data.as_ref().len())
         });
 
         Hash(hash.try_into().unwrap())
     }
+
+    /// Calculates Blake2b-256 digest over given vector of bytes
+    pub fn blake2b_256_hash(data: impl AsRef<[u8]>) -> Hash {
+        let hash = copy_buffer(unsafe {
+            crypto_utils::crypto_utils_blake2b_256_hash(data.as_ref().as_ptr(), data.as_ref().len())
+        });
+
+        Hash(hash.try_into().unwrap())
+    }
+
+    /// Performs Ed25519 signature verification.
+    pub fn ed25519_verify(
+        message: impl AsRef<[u8]>,
+        public_key: impl AsRef<Ed25519PublicKey>,
+        signature: impl AsRef<Ed25519Signature>,
+    ) -> bool {
+        let public_key: Vec<u8> = scrypto_encode(public_key.as_ref()).unwrap();
```

Comment:
As mentioned, I'm not quite sure why we need to scrypto encode for these new methods, and can't pass this through as bytes, e.g. via:
```rust
let public_key: &[u8] = public_key.as_ref().as_ref();
let signature: &[u8] = signature.as_ref().as_ref();
```

Reply:
Reworked